src/inmanta/agent/handler.py

```diff
@@ -1069,53 +1071,22 @@ class HandlerNotAvailableException(Exception):
     """
 
 
-class HandlerLogger(typing.Protocol):
-    """
-    This protocol defines a part of the interface of the inmanta HandlerContext object
-    that can be used to log messages to the server.  It is used in different helper
-    functions across a handler code base.
-
-    When such helper class is used in another context than an agent, they can create
-    another object than the HandlerContext, which simply implements those helper methods,
-    and still be able to use the helper method.
-
-    """
-
-    def debug(self, msg: str, *args: object, **kwargs: object) -> None:
-        ...
-
-    def info(self, msg: str, *args: object, **kwargs: object) -> None:
-        ...
-
-    def warning(self, msg: str, *args: object, **kwargs: object) -> None:
-        ...
-
-    def error(self, msg: str, *args: object, **kwargs: object) -> None:
-        ...
-
-    def exception(self, msg: str, *args: object, exc_info: bool = True, **kwargs: object) -> None:
-        ...
-
-    def critical(self, msg: str, *args: object, **kwargs: object) -> None:
-        ...
-
-
-class LoggerWrapper:
+@stable_api
+class KwargsLogger(IBasicLogging):
     """
-    This class implements the HandlerLogger interface
-
-    It can be used in the tests of a module to be able to use the handler helpers, outside of the context of an agent.
+    This class implements the IBasicLogging interface and is a standalone wrapper
+    around a logging.Logger to facilitate logging of keyword arguments.
```

Comment:
Here too I wouldn't focus on the kwargs aspect. The purpose is more to offer a simple standalone logger compatible with the handler one. Basically what you put in the rest of the docstring so that's good.